### PR TITLE
[test][sanity] add sanity test for dataclasses package

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
- Adding sanity test for dataclasses pip package. The dataclasses package is included as a part of python framework starting py37. 

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: dlc_developer_config.toml [hunk 1]
```diff
@@ -24,7 +24,7 @@ do_build = true
 efa_tests = false
 sanity_tests = true
 sagemaker_tests = false
-ecs_tests = true
-eks_tests = true
-ec2_tests = true
+ecs_tests = false
```
Comment (tejaschumbalkar) [inline dlc_developer_config.toml:27]: temp changes.

File: test/dlc_tests/sanity/test_pre_release.py [hunk 2]
```diff
@@ -311,6 +323,36 @@ def test_dependency_check_neuron(neuron, ec2_connection):
...
+    """
+    Ensure there is no dataclasses pip package is installed for python 3.7 and above version.
+    :param image: ECR image URI
+    """
+    ctx = Context()
+    pip_package = "dataclasses"
+
+    container_name = get_container_name("dataclasses-check", image)
+
+    python_version = re.findall(r'py(\d\.?\d?)', image)[0]
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_pre_release.py:337]: can you use test_utils.get_python_version_from_image_uri here?

File: test/dlc_tests/sanity/test_pre_release.py [hunk 3]
```diff
@@ -311,6 +323,36 @@ def test_dependency_check_neuron(neuron, ec2_connection):
...
+        output = run_cmd_on_container(
+            container_name, ctx, f"pip show {pip_package}", warn=True)
+
+        if output.return_code == 0:
+            pytest.fail(
+                f"{pip_package} package exists in the DLC image {image} that has py{python_version} version which is greater than py36 version")
+        else:
+            LOGGER.info(
+                f"{pip_package} package does not exists in the DLC image {image}")
+    else:
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_pre_release.py:351]: nit: you can probably add a pytest.skip here so that the skip reason shows up in the summary as "Skipped" instead of "Passed"

File: test/dlc_tests/sanity/test_pre_release.py (hunk 2 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/sanity/test_pre_release.py]: ack

**Review by arjkesh — APPROVED** [review]
